src/components/Card.js

```diff
@@ -1,40 +1,56 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    _handleLikeButton,
+    handleCardLike,
+    handleCardDislike
+  ) {
+    this._name = data.name;
+    this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteCard = handleDeleteCard;
+    this._id = data._id;
+    this._handleLikeButton = _handleLikeButton;
+    this.handleLike = handleCardLike;
+    this.handleDislike = handleCardDislike;
+    this._isLiked = data.isLiked;
+  }
+  getCardId() {
+    return this._id;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
     //
     this._cardElement
       .querySelector(".card__trash-button")
       .addEventListener("click", () => {
-        this._handleDeleteCard();
+        this._handleDeleteCard(this);
       });
+    //
+    this._cardElement
+      .querySelector(".card__like-button")
```

Comment:
You've found the like button in line 66 as `this._likeButton`. Please, use it here as well


`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.